Frameworks/include/CoreTextInternal.h

```diff
@@ -32,38 +36,49 @@
 
 @interface _CTFrameSetter : NSObject {
 @public
-    idretaintype(_CTTypesetter) _typesetter;
+    StrongId<_CTTypesetter> _typesetter;
 }
 @end
 
 @interface _CTFrame : NSObject {
 @public
     _CTFrameSetter* _frameSetter;
     CGRect _frameRect;
+    StrongId<NSMutableArray> _lines;
+
+    // TODO::
+    // Do we need these anymore?
     CGSize _totalSize;
-    idretaintype(NSMutableArray) _lines;
     std::vector<CGPoint> _lineOrigins;
 }
 @end
 
 @interface _CTLine : NSObject {
 @public
-    NSRange _strRange;
+    _CTTypesetter* _typeSetter;
+    CFRange _strRange;
+    CGPoint _lineOrigin;
+    StrongId<NSMutableArray> _runs;
+
+    // TODO::
+    // Do we need these anymore?
     CGFloat _width;
     CGFloat _ascent, _descent, _leading;
-    idretaintype(NSMutableArray) _runs;
 }
 @end
 
 @interface _CTRun : NSObject {
 @public
-    idretaintype(NSMutableDictionary) _attributes;
+    StrongId<NSMutableDictionary> _attributes;
     CFRange _range;
     float _xPos;
-    idretaintype(NSString) _stringFragment;
-    std::vector<CGPoint> _glyphOrigins;
+    float _yPos;
+    StrongId<NSString> _stringFragment;
     std::vector<CGSize> _glyphAdvances;
-    std::vector<WORD> _characters;
+
+    // TODO::
+    // Do we need these anymore?
+    std::vector<CGPoint> _glyphOrigins;
```

Comment:
Not sure if I understand this comment. I will sync with you offline

---
In reply to: [77261301](https://github.com/Microsoft/WinObjC/pull/950#discussion_r77261301) [](ancestors = 77261301)